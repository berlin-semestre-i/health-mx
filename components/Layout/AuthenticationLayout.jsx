import React from 'react'
import { Grid, Container } from 'semantic-ui-react'
import styled from 'styled-components'
import media from 'styled-media-query'

const AuthenticationLayout = ({ children }) => (
  <AuthenticationGrid>
    <ImageColumn className="image-container" width={9}/>
    <FormColumn mobile={16} tablet={7} computer={7}>
      <FormContainer>
        { children }
      </FormContainer>
    </FormColumn>
  </AuthenticationGrid>
)

export default AuthenticationLayout

const AuthenticationGrid = styled(Grid)`
  && {
    height: 100vh;
    margin: 0;
    margin-left: -84px;
  }
`
const ImageColumn = styled(Grid.Column)`
  ${media.greaterThan('large')`
    &&&.image-container {
      padding: 0;
      background-image: url("../../static/images/enfermera.png");
      background-repeat: no-repeat;
      background-size: 100%;
    }
  `}

  ${media.lessThan('large')`
    &&&.image-container {
      display: none;
    }
  `}
`

const FormColumn = styled(Grid.Column)`
  & {
    margin: auto;
  }
`

const FormContainer = styled(Container)`
  ${media.lessThan('large')`
    & {
      padding: 0 1rem;
    }
  `}

  ${media.greaterThan('large')`
    & {
      padding: 0 4.5rem;
    }
  `}

  &&& .field, &&& .button {
    margin-bottom: 2em;
  }

  &&& .field label {
    font-size: 16px;
    font-weight: normal;
  }

  &&& .field label:after {
    display: none;
  }

  &&& img {
    max-width: 65%;
    margin-bottom: 3rem;
  }
`