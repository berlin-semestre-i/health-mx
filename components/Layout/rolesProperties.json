{
  "medic": {
    "placeholder": "Buscar derechohabiente o tarea",
    "background": "#3d5170"
  },
  "beneficiary": {
    "placeholder": "Buscar tarea o servicio",
    "background": "#0057A3"
  },
  "nurse": {
    "placeholder": "Buscar derechohabiente",
    "background": "#01493D"
  },
  "admin": {
    "placeholder": "Buscar usuario",
    "background": "#561A9B"
  },
  "error": {
    "placeholder": "Error: you have no role",
    "background": "#F65D61"
  }
}