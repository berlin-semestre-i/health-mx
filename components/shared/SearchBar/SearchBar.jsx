import React, { PureComponent } from 'react'
import { Input } from 'semantic-ui-react'
import styled from 'styled-components'

const Searchbar = styled(Input)`
  &&& {
    width: 100%;
  }

  &&&>input {
    border-radius: 0;
    height: 70px;
    border: none;
    box-shadow: 0 4px 10px -1px #e3e5e8;
    -webkit-box-shadow: 0 4px 10px -1px #e3e5e8;
  }
`

class SearchBar extends PureComponent {
  render() {
    return(
      <Searchbar icon="search" iconPosition="left" placeholder="Buscar derechohabiente o tarea"/>
    )
  }
}

export default SearchBar