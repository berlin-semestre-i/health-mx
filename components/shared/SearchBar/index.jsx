import React from 'react'
import { Input } from 'semantic-ui-react'
import styled from 'styled-components'

const SearchBar = ({ placeholder }) => (
  <Searchbar icon="search" iconPosition="left" placeholder={placeholder}/>
)

export default SearchBar

const Searchbar = styled(Input)`
  &&& {
    width: 100%;
  }

  &&&>input {
    font-size: 18px;
    font-weight: 300;
    border-radius: 0;
    height: 70px;
    border: none;
    box-shadow: 0 4px 10px -1px #e3e5e8;
    -webkit-box-shadow: 0 4px 10px -1px #e3e5e8;
  }
`
