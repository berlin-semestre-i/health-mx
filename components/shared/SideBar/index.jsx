import React from 'react'
import Link from 'next/link'
import { withRouter } from 'next/router'
import { Menu, Image, Icon, Popup } from 'semantic-ui-react'
import styled from 'styled-components'
import { getPropertiesFromRole } from '../../../utils/auth'
import { defaultManAvatar, defaultWomanAvatar } from '../../../utils/constants'

class SideBar extends React.PureComponent {
  render() {
    const { userRole, router, userAvatar, userGender } = this.props
    const roleProperties = getPropertiesFromRole(userRole)
    const genderDefaultAvatar = userGender === 'male' ? defaultManAvatar : defaultWomanAvatar
    const route = router.route

    return (
      <SideMenu
        name="sideMenu"
        vertical
        fixed="left"
        background={roleProperties.background}
        icon
      >
        <Link href={`/${userRole}`}>
          <HealthImage src="../../../static/images/Health-Bar-Logo.png"/>
        </Link>
        <Link href={`/${userRole}`}>
          <Menu.Item
            name="home"
            active={`/${userRole}` === route}
          >
            <Popup
              flowing
              inverted
              size="tiny"
              position="top right"
              horizontalOffset={12}
              trigger = {
                <Icon name="home" size="large"/>
              }
              content="Inicio"
            />
          </Menu.Item>
        </Link>
        { roleProperties.menuItems.map(item => (
          <Link href={`/${userRole}${item.address}`} key={item.id}>
            <Menu.Item
              name={item.name}
              active={`/${userRole}${item.address}` === route}
            >
              <Popup
                flowing
                inverted
                size="tiny"
                position="top right"
                horizontalOffset={12}
                trigger = {
                  <Icon name={item.icon} size="large"/>
                }
                content={item.description}
              />
            </Menu.Item>
          </Link>
        ))}
        <Notifications name="notifications">
          <Popup
            flowing
            inverted
            size="tiny"
            position="top right"
            horizontalOffset={12}
            trigger = {
              <Icon name="bell" size="large"/>
            }
            content="Notificaciones"
          />
        </Notifications>
        <Tips name="tips">
          <Popup
            flowing
            inverted
            size="tiny"
            position="top right"
            horizontalOffset={12}
            trigger = {
              <Icon name="question circle" size="large"/>
            }
            content="Ayuda"
          />
        </Tips>
        <Avatar
          name="userAvatar"
          src={userAvatar ? userAvatar : genderDefaultAvatar}
          circular
        />
      </SideMenu>
    )
  }
}

export default withRouter(SideBar)

const HealthImage = styled(Image)`
  && {
    margin: 15px 21px 36px 21px;
    width: 42px;
    height: 40px;
    cursor: pointer;
  }
`

const SideMenu = styled(Menu)`
  &&&& {
    display: flex;
    border: none;
    -webkit-box-shadow: 0 0 10px 1px #e7e9ed;
    box-shadow: 0 0 10px 1px #e7e9ed;
    background-color: ${props => props.background};
  }
`

const Avatar = styled(Image)`
  &&&& {
    background-color: #FFFFFF;
    width: 48.8px;
    height: 48.8px;
    margin: 35.6px 18px 50.6px 21px;
    align-self: flex-end;
    border: solid 2px #FFFFFF;
  }
`

const Notifications = styled(Menu.Item)`
  &&&& {
    margin-top: auto;
    align-self: flex-end;
  }
`

const Tips = styled(Menu.Item)`
  &&&& {
    margin-bottom: 0;
    align-self: flex-end;
  }
`