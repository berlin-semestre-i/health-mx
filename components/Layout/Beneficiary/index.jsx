import React, { Fragment } from 'react'
import SideBar from '../../shared/SideBar/SideBar'
import SearchBar from '../../shared/SearchBar/SearchBar'
import { CustomContainer, MainContainer } from '../../../static/styledComponents'

const Layout = ({ children }) => (
  <Fragment>
    <SideBar className="beneficiary"/>
    <CustomContainer>
      <SearchBar placeholder="Buscar tarea o servicio"/>
      <MainContainer>
        { children }
      </MainContainer>
    </CustomContainer>
  </Fragment>
)

export default Layout