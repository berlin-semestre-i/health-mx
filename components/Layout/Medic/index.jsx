import React from 'react'
import { Container } from 'semantic-ui-react'
import styled from 'styled-components'
import SideBar from '../../shared/SideBar/SideBar'
import SearchBar from '../../shared/SearchBar/SearchBar'

const CustomContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
  }
`

const MainContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
    padding: 3rem 3rem 0;
  }
`

const Layout = ({ children }) => (
  <React.Fragment>
    <SideBar background="#3d5170"/>
    <CustomContainer>
      <SearchBar />
      <MainContainer>
        { children }
      </MainContainer>
    </CustomContainer>
  </React.Fragment>
)

export default Layout