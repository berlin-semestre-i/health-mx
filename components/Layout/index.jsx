import React, { Fragment } from 'react'
import { Container } from 'semantic-ui-react'
import styled from 'styled-components'
import SideBar from '../shared/SideBar/SideBar'
import SearchBar from '../shared/SearchBar/SearchBar'
import { medic, beneficiary, nurse, error, admin } from './rolesProperties.json'

const getPropertiesFromRole = (userRole) => {
  if ( userRole === 'medic' ) {
    return medic
  }
  else if ( userRole === 'beneficiary' ) {
    return beneficiary
  }
  else if ( userRole === 'nurse' ) {
    return nurse
  }
  else if ( userRole === 'admin' ) {
    return admin
  }
  else {
    return error
  }
}

const Layout = ({ children, userRole }) => {
  const layoutProperties = getPropertiesFromRole(userRole)
  return (
    <Fragment>
      <SideBar background={layoutProperties.background}/>
      <GlobalContainer>
        <SearchBar placeholder={layoutProperties.placeholder}/>
        <MainContainer>
          { children }
        </MainContainer>
      </GlobalContainer>
    </Fragment>
  )
}

export default Layout

const GlobalContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
  }
`

const MainContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
    padding: 3rem 3rem 0;
  }
`
