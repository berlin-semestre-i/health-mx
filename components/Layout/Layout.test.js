import React from 'react'
import { shallow, mount } from 'enzyme'
import Layout from './index'
import SearchBar from '../shared/SearchBar/SearchBar'
import SideBar from '../shared/SideBar/SideBar'
import { medic, beneficiary, nurse, admin, error } from './rolesProperties.json'
import 'jest-styled-components'

const defaultRole = 'medic'

describe('Layout', () => {
  it('renders', () => {
    const component = shallow(<Layout userRole={defaultRole}/>)
    expect(component).toMatchSnapshot()
  })

  it('should contain a SearchBar and a SideBar', () => {
    const wrapper = mount(<Layout userRole={defaultRole}/>)
    expect(wrapper.contains(<SearchBar placeholder={medic.placeholder}/>)).toEqual(true)
    expect(wrapper.contains(<SideBar background={medic.background}/>)).toEqual(true)
    wrapper.unmount()
  })

  it('should have medic properties when medic role', () => {
    const wrapper = mount(<Layout userRole="medic"/>)
    expect(wrapper.contains(<SearchBar placeholder={medic.placeholder}/>)).toEqual(true)
    expect(wrapper.contains(<SideBar background={medic.background}/>)).toEqual(true)
    expect(wrapper).toHaveStyleRule(
      'background-color',
      medic.background,
      {
        modifier: '&&&&',
      }
    )
    wrapper.unmount()
  })

  it('should have beneficiary properties when beneficiary role', () => {
    const wrapper = mount(<Layout userRole="beneficiary"/>)
    expect(wrapper.contains(<SearchBar placeholder={beneficiary.placeholder}/>)).toEqual(true)
    expect(wrapper.contains(<SideBar background={beneficiary.background}/>)).toEqual(true)
    expect(wrapper).toHaveStyleRule(
      'background-color',
      beneficiary.background,
      {
        modifier: '&&&&',
      }
    )
    wrapper.unmount()
  })

  it('should have nurse properties when nurse role', () => {
    const wrapper = mount(<Layout userRole="nurse"/>)
    expect(wrapper.contains(<SearchBar placeholder={nurse.placeholder}/>)).toEqual(true)
    expect(wrapper.contains(<SideBar background={nurse.background}/>)).toEqual(true)
    expect(wrapper).toHaveStyleRule(
      'background-color',
      nurse.background,
      {
        modifier: '&&&&',
      }
    )
    wrapper.unmount()
  })

  it('should have admin properties when admin role', () => {
    const wrapper = mount(<Layout userRole="admin"/>)
    expect(wrapper.contains(<SearchBar placeholder={admin.placeholder}/>)).toEqual(true)
    expect(wrapper.contains(<SideBar background={admin.background}/>)).toEqual(true)
    expect(wrapper).toHaveStyleRule(
      'background-color',
      admin.background,
      {
        modifier: '&&&&',
      }
    )
    wrapper.unmount()
  })

  it('should have error properties when no existing role', () => {
    const wrapper = mount(<Layout userRole="nonExistingRole"/>)
    expect(wrapper.contains(<SearchBar placeholder={error.placeholder}/>)).toEqual(true)
    expect(wrapper.contains(<SideBar background={error.background}/>)).toEqual(true)
    expect(wrapper).toHaveStyleRule(
      'background-color',
      error.background,
      {
        modifier: '&&&&',
      }
    )
    wrapper.unmount()
  })
})