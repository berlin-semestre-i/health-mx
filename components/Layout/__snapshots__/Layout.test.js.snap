// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout
    userRole="medic"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <SideBar
          background="#3d5170"
        />,
        <ForwardRef>
          <SearchBar
            placeholder="Buscar derechohabiente o tarea"
          />
          <ForwardRef />
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "background": "#3d5170",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <SearchBar
              placeholder="Buscar derechohabiente o tarea"
            />,
            <ForwardRef />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "placeholder": "Buscar derechohabiente o tarea",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "Layout__MainContainer-sc-1tdk3as-1",
                "isStatic": true,
                "rules": Array [
                  "&&&{margin:0;width:100%;padding:3rem 3rem 0;}",
                ],
              },
              "displayName": "Layout__MainContainer",
              "handledProps": Array [
                "as",
                "children",
                "className",
                "content",
                "fluid",
                "text",
                "textAlign",
              ],
              "render": [Function],
              "styledComponentId": "Layout__MainContainer-sc-1tdk3as-1",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Layout__GlobalContainer-sc-1tdk3as-0",
            "isStatic": true,
            "rules": Array [
              "&&&{margin:0;width:100%;}",
            ],
          },
          "displayName": "Layout__GlobalContainer",
          "handledProps": Array [
            "as",
            "children",
            "className",
            "content",
            "fluid",
            "text",
            "textAlign",
          ],
          "render": [Function],
          "styledComponentId": "Layout__GlobalContainer-sc-1tdk3as-0",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <SideBar
            background="#3d5170"
          />,
          <ForwardRef>
            <SearchBar
              placeholder="Buscar derechohabiente o tarea"
            />
            <ForwardRef />
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "background": "#3d5170",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <SearchBar
                placeholder="Buscar derechohabiente o tarea"
              />,
              <ForwardRef />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "placeholder": "Buscar derechohabiente o tarea",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "Layout__MainContainer-sc-1tdk3as-1",
                  "isStatic": true,
                  "rules": Array [
                    "&&&{margin:0;width:100%;padding:3rem 3rem 0;}",
                  ],
                },
                "displayName": "Layout__MainContainer",
                "handledProps": Array [
                  "as",
                  "children",
                  "className",
                  "content",
                  "fluid",
                  "text",
                  "textAlign",
                ],
                "render": [Function],
                "styledComponentId": "Layout__MainContainer-sc-1tdk3as-1",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "Layout__GlobalContainer-sc-1tdk3as-0",
              "isStatic": true,
              "rules": Array [
                "&&&{margin:0;width:100%;}",
              ],
            },
            "displayName": "Layout__GlobalContainer",
            "handledProps": Array [
              "as",
              "children",
              "className",
              "content",
              "fluid",
              "text",
              "textAlign",
            ],
            "render": [Function],
            "styledComponentId": "Layout__GlobalContainer-sc-1tdk3as-0",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
